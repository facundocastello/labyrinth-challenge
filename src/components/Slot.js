import styled from 'styled-components';

const typeStyles = {
  'end': '#85EB85',
  'start': '#FFFF07',
  'blocked': 'lightgray',
  'empty': 'white'
};

export default styled.div`
  width:30px;
  height:30px;
  border: 1px solid black;
  background-color: ${( {type} ) => typeStyles[type] }
`;