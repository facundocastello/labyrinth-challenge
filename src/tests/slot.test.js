import React from 'react';
import Slot from '../components/Slot';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

describe( 'Slot Block', () => {
  it( 'renders Slot', () => {
    const tree = renderer.create(
      <Slot />
    ).toJSON();
    expect( tree ).toMatchSnapshot();
  } );
} );