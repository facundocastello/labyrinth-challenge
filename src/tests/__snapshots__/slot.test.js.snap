// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slot Block renders Slot 1`] = `
.c0 {
  width: 30px;
  height: 30px;
  border: 1px solid black;
}

<div
  className="c0"
/>
`;
